src/api/app/models/bs_request_action.rb

```diff
@@ -798,6 +781,23 @@ def set_sourceupdate_default(user)
 
   private
 
+  def check_patchinfo(pkg)
+    pkg.project.repositories.each do |repo|
+      firstarch = repo.architectures.first
+      next unless firstarch
+
+      # skip excluded patchinfos
+      status = pkg.project.project_state.search("/resultlist/result[@repository='#{repo.name}' and @arch='#{firstarch.name}']").first
+
+      next if status && (s = status.search("status[@package='#{pkg.name}']").first) && s.attributes['code'].value == 'excluded'
+
+      raise BuildNotFinished, "patchinfo #{pkg.name} is broken" if s.attributes['code'].value == 'broken'
+
+      check_maintenance_release(pkg, repo, firstarch)
+    end
+    true
```

Comment:
This always returns true. It should only return true if it finds a patchinfo.

Reply:
not true. did you check the original code? it will assign true if no exception is raised.

Reply:
Fixed. Thanks!